lib/screens/main_screen/main_screen.dart

```diff
@@ -2,72 +2,95 @@ import 'package:carousel_slider/carousel_slider.dart';
-            ),
             SliverPadding(
               padding: const EdgeInsets.all(20),
-              sliver: SliverGrid.count(
-                crossAxisCount: 2,
-                childAspectRatio: 173 / 224,
-                crossAxisSpacing: 20,
-                mainAxisSpacing: 20,
-                children: const [
-                  DeptWidget(
-                    image: "aids",
-                    department: DepartmentEnum.aids,
-                  ),
-                  DeptWidget(
-                    image: "extc",
-                    department: DepartmentEnum.extc,
-                  ),
-                  DeptWidget(
-                    image: "cs",
-                    department: DepartmentEnum.cs,
-                  ),
-                  DeptWidget(
-                    image: "it",
-                    department: DepartmentEnum.it,
-                  ),
-                  DeptWidget(
-                    image: "biomed",
-                    department: DepartmentEnum.biomed,
-                  ),
-                  DeptWidget(
-                    image: "biotech",
-                    department: DepartmentEnum.biotech,
-                  ),
-                  DeptWidget(
-                    image: "chem",
-                    department: DepartmentEnum.chem,
-                  ),
-                ],
+              sliver: SliverList(
+                delegate: SliverChildListDelegate(
+                  [
+                    Container(
+                      width: _size.width * 0.9,
+                      decoration: BoxDecoration(
+                        color: _theme.primaryColor,
+                        borderRadius: BorderRadius.circular(15.0),
+                        border: Border.all(
+                          color: _theme.primaryColorLight,
+                          width: 1,
+                          style: BorderStyle.solid,
+                        ),
+                        boxShadow: [_boxshadow],
+                      ),
+                      child: DatePicker(
+                        DateTime.now(),
+                        monthTextStyle: _theme.textTheme.subtitle2!,
+                        dayTextStyle: _theme.textTheme.subtitle2!,
+                        dateTextStyle: _theme.textTheme.subtitle2!,
+                        initialSelectedDate: DateTime.now(),
+                        selectionColor: Colors.blue,
+                        daysCount: 7,
+                      ),
+                    ),
+                    const SizedBox(
+                      height: 20.0,
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.symmetric(
+                        horizontal: 4.0,
+                        vertical: 15.0,
+                      ),
+                      child: SizedBox(
+                        height: _size.height * 0.6,
+                        width: _size.width,
+                        child: ListView.builder(
```

Comment:
Don't use ListView inside a CustomScrollView

Better split the implementation in 2 part

1. The normal components (SliverToBox)
2. SliverList for long lists